Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -564,9 +564,9 @@ bool TryInitialize() {
 
             [configuration release];
 
-            id<MTLLibrary> lib = CompileShader(metalDevice, shaderSource);
-            id<MTLFunction> vertexFunction = [lib newFunctionWithName:@"vertexShader"];
-            id<MTLFunction> fragmentFunction = [lib newFunctionWithName:@"fragmentShader"];
+            lib = CompileShader(metalDevice, shaderSource);
+            vertexFunction = [lib newFunctionWithName:@"vertexShader"];
+            fragmentFunction = [lib newFunctionWithName:@"fragmentShader"];
```

Comment:
Discussed offline, but keep these as local variables and just `release` them after they are assigned to the `pipelineStateDescriptor` (e.g. down on line 577).